Scripts/Spice.py

```diff
@@ -17,14 +19,185 @@ def compute_view_frustrum(instrument):
+    """
+    Loaded from SPK file
+    Returns the position of a body relative to another observing body
+
+    Example: body_position('Cassini', 595166468.1826606, 'J2000', 'NONE', 'SATURN BARYCENTER')
+
+    https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/spkpos_c.html
+    :param mainBody: the name of target body
+    :param etTime: time in ET format (look at convert_UTC_to_ET)
+    :param referenceFrame: string referring to the reference frame of the output position vector
+    :param correction:  Abberation correction flag
+    :param observingBody: Name of observing body
+    :return: Position of target in a 3D vector
+    :return: light time between the objects as a double
+    """
+
+    return spice.spkpos(mainBody, etTime, referenceFrame, correction, observingBody)
+
+def find_frame_transformation(fromObject: str, toObject:str, etTime: float):
+    """
+    Returns the transformation matrix of how the frame is moving
+
+    Example: find_frame_transformation ( 'IAU_EARTH', 'J2000', ET  )
+
+    ftp://naif.jpl.nasa.gov/pub/naif/toolkit_docs/FORTRAN/spicelib/pxform.html
+
+    :param fromObject: name of FK of which we are coming from
+    :param toObject: name of FK of which we are going to
+    :param etTime: Epoch of the rotation matrix
+    :return: ET is the amount the position has transformed (magnitude?)
+    :return: The 3x3 transformation matrix M, where (x,y,z)^T *M = (x2,y2,z2)^T
+    """
+
+    return spice.pxform(fromObject, toObject, etTime)
+
+##############
+# DSK Kernel #
+##############
+
+def find_ray_surface_intercept(method: str, target: str, etTime: float, fixedReference:str, correction: str,
+                  observerName: str, directionReference:str, directionVector: list):
+    """
+    Finds where a ray intercepts a surface
+
+    https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/FORTRAN/spicelib/sincpt.html
+
+    :param method: computational method name such as 'ELLIPSOID'
+    :param target: name of the target body
+    :param etTime: et time of the frame
+    :param fixedReference: name of the FK that the target body is fixed to
+    :param correction: Aberration correction
+    :param observerName: the name of the observer taking the frame
+    :param directionReference: name of FK that is the reference frame of which the ray's direction is
+    :param directionVector: Ray direction vector that is from the observer
+    :return: The closest point on the surface that the ray intersects in Cartesion coordinates
+    :return: The intercept epoch, or the distance between the observer and the intercept point
+    :return: Vector from the observer's position at ET to the corrected position of surface intersect point
+    :return: Boolean value which expresses if the ray intersects
+    """
+
+    return spice.sincpt(method, target, etTime, fixedReference, correction, observerName, directionReference,
+                        directionVector)
+
+def find_fov(instrumentID: int, maxReturn: int):
```

Comment:
What would be a sensible default value for `maxReturn`?

Reply:
I haven't found anything above 4 yet and we only run into errors when the maxReturn is too low for the instrument so we can probably high ball it and go for 6-10ish?

Reply:
```suggestion
def find_fov(instrumentID: int, maxReturn: int = 10):
```